File: js/task1/S1_VAR_06.js
// S1_VAR_06

function classifyNumberLike(x) {
    if (Number.isNaN(x)) {
        return "nan";
    }
    if (typeof x === "number") {
        return "number";
    }
    return "not-a-number";
}

console.log(classifyNumberLike(NaN));
console.log(classifyNumberLike(123));
console.log(classifyNumberLike("123"));

File: python/task1/S1_VAR_06.py
# S1_VAR_06

def to_int_or_none(s):
    try:
        if s is None:
            return None
        return int(s)
    except ValueError:
        return None

print(to_int_or_none("12"))
print(to_int_or_none(" 12 "))
print(to_int_or_none("12x"))
print(to_int_or_none(""))
print(to_int_or_none(None))